app/(gcforms)/[locale]/(user authentication)/profile/page.tsx

```diff
@@ -19,13 +19,8 @@ export async function generateMetadata({
 export default async function Page({ params: { locale } }: { params: { locale: string } }) {
   const { session, ability } = await authCheckAndRedirect();
 
-  const hasPublishPrivilege = checkPrivilegesAsBoolean(ability, [
-    {
-      action: "update",
-      subject: { type: "FormRecord", object: { users: [{ id: session.user.id }] } },
-      field: "isPublished",
-    },
-  ]);
+  // Check is a user can update at least one FormRecord and has the privilege to publish
+  const hasPublishPrivilege = ability.can("update", "FormRecord", "isPublished");
```

Comment:
Using the built in `ability.can` in this instance because we are asking the question if a user has the ability to update the `isPublished` field on at least one FormRecord object.